framework/aster-frame/src/arch/x86/irq.rs

```diff
@@ -25,6 +22,14 @@ pub(crate) fn init() {
         });
     }
     IRQ_LIST.call_once(|| list);
+    CALLBACK_ID_ALLOCATOR.call_once(|| Mutex::new(IdAlloc::with_capacity(256)));
+    IRQ_ALLOCATOR.call_once(|| {
+        let mut id_alloc = IdAlloc::with_capacity(256);
```

Comment:
I think a new constructor `IdAlloc::with_capacity_and_min_id` should be added instead. Using this constructor is more elegant than calling `alloc` or `reserve` repeatedly.

Reply:
I'd like to argue against the method `with_capacity_and_min_id` because we found only one usage for this: the first 32 irq numbers are reserved by Intel for specific purposes. And we may not expect other ID allocation rules that require first few IDs not being able to be allocated. If a range of IDs with a high lower bound is required, the ID allocator user should have a linear mapping to reduce the overhead brought by the bitvec.

So I think the less elegant initialization should be allowed here, as it is only initialized once and signifies the reservation of irq numbers. And the new constructor may not be generic enough to be a method of the allocator.

Reply:
This is a fair counter argument. But at least the loop can be simplified by using `alloc_consecutive`.

```rust
// Reserve the first 32 IRQ lines, which are used for CPU exceptions, not external interrupts
let id_range = id_alloc.alloc_consecutive(32).unwrap();
debug_assert!(id_range == (0..32));
```

Reply:
Well, that means we need to have an extra method `alloc_specific_consecutive(r: Range<usize>)` and `alloc_specific_consecutive(0..32)`.

Use `alloc_consecutive(32)` here would imply that we should allocate in order. That's an extra invariant that is should be ensured on further iterations of the code.

Reply:
Although this is an unwritten contract of `IdAlloc`, we have tested that the property is held by having the `debug_assert`.

Reply:
Ok thanks. Since the initialization is only called once and `alloc_specific` here is `O(1)`, I'd like to keep the loop.